source/DataTracker.mc

```diff
@@ -3,112 +3,129 @@ using Toybox.ActivityMonitor as Act;
 //! Class that holds the data that has to be
 //! valculated and is displayed in the UI.
 class DataTracker {
-	//! Points to win a set
-	hidden var SCORE_FOR_SET = 11;
-	//! Number of steps done during the activity
-	hidden var numberOfSteps;
-	//! Number of calories burned during the activity
-	hidden var numberOfCalories;
-	//! Number of steps done when the activity started
-	hidden var initialSteps;
-	//! Amount of calories burnt until the activity started
-	hidden var initialCalories;
-	//! Session used to record the activity
-	hidden var session;
-	//! Sets tracker
-	hidden var setsTracker;
-	
-	//! Constructor
-	function initialize() {
-		session = new Session();
-		restart();
-	}
-	
-	//! Restarts all the data field to make it ready
-	//! to record a new session
-	function restart() {
-		numberOfSteps = 0;
-		numberOfCalories = 0;
+    //! Points to win a set
+    hidden var SCORE_FOR_SET = 89;
+    //! Sets to win a match
+    hidden var SCORE_FOR_MATCH = 88;
+    //! Number of steps done during the activity
+    hidden var numberOfSteps;
+    //! Number of calories burned during the activity
+    hidden var numberOfCalories;
+    //! Number of steps done when the activity started
+    hidden var initialSteps;
+    //! Amount of calories burnt until the activity started
+    hidden var initialCalories;
+    //! Session used to record the activity
+    hidden var session;
+    //! Sets tracker
+    hidden var setsTracker;
+
+    //! Constructor
+    function initialize(setMaxScore, setTotalSets) {
+        session = new Session();
+        SCORE_FOR_SET = setMaxScore;
+        SCORE_FOR_MATCH = setTotalSets;
+        restart();
+    }
+
+    //! Restarts all the data field to make it ready
+    //! to record a new session
+    function restart() {
+        numberOfSteps = 0;
+        numberOfCalories = 0;
         var activityInfo = Act.getInfo();
         initialSteps = activityInfo.steps;
         initialCalories = activityInfo.calories;
-		if (session != null && session.isRecording()) {
-			var gameScore = setsTracker.getGameScore();
-			session.saveGameScore(gameScore[0], gameScore[1]);
-			session.stop();
-		}
-		setsTracker = new SetsTracker();
-	}
-	
-	//! Updates the calculated values taken from
-	//! the activity info data
-	function update() {
-		var activityInfo = Act.getInfo();
+        if (session != null && session.isRecording()) {
+            var gameScore = setsTracker.getGameScore();
+            session.saveGameScore(gameScore[0], gameScore[1]);
+            session.stop();
+        }
+        setsTracker = new SetsTracker(SCORE_FOR_SET, SCORE_FOR_MATCH);

```

Comment:
This breaks encapsulation a little bit. The Data Tracker shouldn't know about the details of how sets and matches logic work. For now, let's keep it like this, I will change it later.